### ISSUE TITLE
Mesh Export, option to set bounds from current position, camera, etc

### ISSUE THREAD

**Issue by mancoast** [buddhi1980/mandelbulber2#451]
I would like an easy to operate control that automatically configures the mesh export bounds.
The parameters for the export dialog are approximated from the current fractal view.

**Comment by buddhi1980** [comment]
Maybe it's doable. How would you define the bounds? Everything which is in front of the camera? Objects which are not farther than...? or in another way..?

**Comment by mancoast** [comment]
Exactly! I really like your idea for implementation based on what is in front of the camera to intuitively export subsections of interest.

**Comment by buddhi1980** [comment]
@mancoast It's not enough precise answer. Just imagine that you are sitting in the room by the desk with computer. You would like to make 3D objects of that what **is in front of you**. You would like to have 3D object of just a keyboard? O maybe including LCD  display? Or maybe the wall? Or maybe entire room? Or everything which is behind the wall? 
If model of the desk, would you like to have only that parts of the desk which are actually visible for you (your field of view is limited).
I hope by this example I have showed you that I need more precise explanation from you.

**Comment by mancoast** [comment]
Imagine the x,y,z bounds are all the same magnitude. The visible window should provide a value for x and y ranges.  How about we simply select a z range that is similar magnitude of x range?

**Comment by zebastian** [comment]
There is now a new mouse click function which can be used to catch an object inside a bounding box:
Wrap limits around target
I also just added a couple of buttons to make the handling of the limits user-friendlier, see under Rendering Engine > Limits

**Comment by JanWerder** [comment]
This is a very nice addition, since you don't have to guess the bounds of a fractal, but a "get limits from current view" type of feature would be even better. When you zoomed into a fractal you don't really have the ability to find out the current bounds, other than hovering your mouse all over the place, just to get some coordinates.

**Comment by buddhi1980** [comment]
"get limits from current view" doesn't make sense. How the program will know how to define bounding box based on actual view? Visible area has pyramid shape. How deep, wide and high it should be?

**Comment by JanWerder** [comment]
Well, that could be parameters. But if the Export could center around the current x,y,z that would help
    	
    	

    	
    
  




On Sun, Dec 2, 2018 at 5:51 PM +0100, "Krzysztof Marczak" <notifications@github.com> wrote:












"get limits from current view" doesn't make sense. How the program will know how to define bounding box based on actual view? Visible area has pyramid shape. How deep, wide and high it should be?

**Comment by mancoast** [comment]
The viewport is composed of camera xyz and target xyz. Use target xyz values for center. Use distance from camera to target for bounds.

**Comment by mancoast** [comment]
Sounds intriguing. Can you provide more details? I will initiate the efforts because I really want this feature for 3D printing objects from deep zooms.
